clients/java/client/pom.xml

```diff
@@ -159,13 +159,6 @@
       <scope>provided</scope>
     </dependency>
 
-    <dependency>
```

Comment:
This dependency was added to resolve error in external task client: ([See conversation](https://github.com/camunda/camunda-bpm-platform/issues/4192#issuecomment-2082075944))
```
Caused by: java.lang.NoClassDefFoundError: javax/xml/bind/JAXBException
```

Above error is a side effect of the shaded dependencies leaking into the build process. The current commit resolves the root cause, so the work around can be removed.